packages/svelte/src/compiler/phases/2-analyze/a11y.js

```diff
@@ -601,21 +601,20 @@ function is_parent(parent, elements) {
  */
 function validate_aria_attribute_value(attribute, name, schema, value) {
 	const type = schema.type;
-	const is_string = typeof value === 'string';
 
 	if (value === null) return;
-	if (value === true) value = 'true'; // TODO this is actually incorrect, and we should fix it
+
+	if (value === true) {
```

Comment:
I think this is too broad - for example for `aria-labbeledby` this warning would be wrong. Looking at the code further below it seems we have existing warnings for boolean aria attributes already, why don't we just use those?

Reply:
What do you mean it would be wrong? `aria-labelledby` can't be empty, it has to be one or more IDs

Reply:
The message would be wrong because it would tell you to set it to `true`. For `aria-describedby` it would be equally confusing. And again I'm wondering why we need a new warning in the first place, it seems the code below has existing handling for incorrect values already, depending on what type they are.

Reply:
We need a new warning because there's no situation in which an empty attribute is valid, and given that elsewhere an empty attribute means 'true' it's confusing to tell people that 'true' is a valid value but their empty attribute isn't. What if we have two warnings — keep as it is here for `boolean` and `tristate` attributes, and for everything else just say 

> '%attribute%' cannot be empty

?

Reply:
I don't think we need any new warnings? Below this check there's a bunch of statements emitting warnings for various cases. For example 
```js
if (type === 'boolean' && value !== 'true' && value !== 'false') {
	w.a11y_incorrect_aria_attribute_type_boolean(attribute, name);
}
```
will automatically handle the invalid boolean value case.
For `type=token` and `type=tokenlist` the if checks need to be slightly adjusted, but other than that this should then handle this just fine?

Reply:
That's what I'm saying — you'd get a warning like

> The value of '%attribute%' must be either 'true' or 'false'

to which a developer might reasonably say 'but `<div aria-hidden>` already means 'true', unaware of the difference between boolean attributes and boolean ARIA attributes.

We could adjust `a11y_incorrect_aria_attribute_type_boolean` to say this...

> The value of '%attribute%' must be either 'true' or 'false'. It cannot be empty

...but then you're no longer narrowing the type of `value` to `string`, which complicates the code further below. It seems like the suggestion above (different warning on empty attributes for boolean/tristate) is just more pragmatic

Reply:
I just realized that we can also assign the empty string to the value instead (that's what it means after all) - then all the code further below doesn't need to be adjusted. And adjusting the warning sounds ok to me.